server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -563,6 +648,11 @@ public ParsedDocument index(IndexItem item) throws IOException {
             Object[] values = item.insertValues();
             for (int i = 0; i < values.length; i++) {
                 Reference reference = columns.get(i);
+                if (writeOids && reference instanceof DynamicReference == false) {
+                    // It's possible to have references with OID after doing a mapping update:
+                    // Empty arrays, arrays with only null values and columns added dynamically into IGNORED object doesn't result in schema update.
+                    assert reference.oid() != COLUMN_OID_UNASSIGNED : "All target columns must have assigned OID on indexing.";
+                }
```

Comment:
Can we move this into a method and assert the method instead? This will remove the if condition checks if assertions are disabled.
```suggestion
                    // It's possible to have references with OID after doing a mapping update:
                    // Empty arrays, arrays with only null values and columns added dynamically into IGNORED object doesn't result in schema update.
                    assert assertExistingOid(reference) : "All target columns must have assigned OID on indexing.";
               
```

```
private boolean assertExistingOid(Reference ref) {
    if (writeOids && reference instanceof DynamicReference == false) {
        return ref.oid() != COLUMN_OID_UNASSIGNED
    }
    return true;
}
```

Reply:
Thanks, applied the suggestion and also updated wrong code comment (instead of "with OID" should be "without OID")